keps/sig-storage/1710-selinux-relabeling/README.md

```diff
@@ -111,153 +137,271 @@ Following conditions must be met:
+
+```go
+// In storage.k8s.io/v1:
+
+// CSIDriverSpec is the specification of a CSIDriver.
+type CSIDriverSpec struct {
+// SELinuxMount specifies if the CSI driver supports "-o context"
+// mount option.
+//
+// When "true", Kubernetes may call NodeStage / NodePublish with "-o context=xyz" mount
+// option. The CSI driver must ensure that all volumes can be mounted with different
+// `-o context` options. This is typical for storage backends that provide volumes
+// as filesystems on block devices or as independent shared volumes.
+// It is task of the CSI driver to add any necessary mount options to allow
+// mounting with `-o context`, for example `nosharesock` for CIFS or
+// `nosharecache` for NFS.
+//
+// When "false", Kubernetes won't pass any special SELinux mount options to the driver.
+// This is typical for volumes that cannot mount a volume with `-o context` mount option.
+//
+// Default is "false".
+SELinuxMount *bool;
+...
+}
+
+// For context:
+type CSIDriver struct {
+Spec CSIDriverSpec
+}
+```
+
+The default value is `false` to ensure backward compatibility.
+
+##### `PodSecurityContext`
+Feature gate `SELinuxChangePolicy`, **proposing alpha in 1.32.**
+
+```go
+// PodSecurityContext holds pod-level security attributes and common container settings.
+// Some fields are also present in container.securityContext.  Field values of
+// container.securityContext take precedence over field values of PodSecurityContext.
+type PodSecurityContext struct {
+// ...
+	// seLinuxChangePolicy defines how the container's SELinux label is applied to all volumes used by the Pod.
+	// It has no effect on nodes that do not support SELinux or when the volume does not support SELinux.
+    // Valid values are "MountOption" and "Recursive". If not specified, "MountOption" is used.
+	// It affects only in-tree iSCSI and FibreChannel volumes, and CSI volumes that announce "SELinuxMount: true" in their CSIDriver instance.
+	// It affects only Pods that have SELinux label set, either in PodSecurityContext or in SecurityContext of all containers.
+    // Note that this field cannot be set when spec.os.name is windows.
+    // +optional
+    SELinuxChangePolicy *PodSELinuxChangePolicy
+}
+
+// PodSELinuxChangePolicy defines how the container's SELinux label is applied to all volumes used by the Pod.
+type PodSELinuxChangePolicy string
+
+const (
+	// Recursive relabeling of all Pod volumes by the container runtime.
+	// This may be slow for large volumes, but allows mixing privileged and unprivileged Pods sharing the same volume on the same node.
+    SELinuxChangePolicyRecursive PodSELinuxChangePolicy = "Recursive"
+	// MountOption mounts all Pod volumes with `-o context` mount option.
+	// This requires all Pods that share the same volume to use the same SELinux label.
+	// It is not possible to share the same volume among privileged and unprivileged Pods.
+    SELinuxChangePolicyMountOption PodSELinuxChangePolicy = "MountOption"
```

Comment:
Given that this will be the behavior otherwise, I'm not sure its necessary to have value. `nil` and `Recursive` seem to cover it.